src/button-group/__tests__/button-group.test.tsx

```diff
@@ -0,0 +1,253 @@
+  await new Promise(resolve => setTimeout(resolve, duration));
+};
+
+const items1: ButtonGroupProps.ItemOrGroup[] = [
+  {
+    type: 'group',
+    text: 'Feedback',
+    items: [
+      { type: 'icon-button', id: 'like', text: 'Like', iconName: 'thumbs-up', popoverFeedback: 'Liked' },
+      {
+        type: 'icon-button',
+        id: 'dislike',
+        disabled: true,
+        text: 'dislike',
+        iconName: 'thumbs-down',
+        popoverFeedback: 'Disliked',
+      },
+    ],
+  },
+  { type: 'icon-button', id: 'copy', iconName: 'copy', text: 'Copy', popoverFeedback: 'Copied' },
+  { type: 'icon-button', id: 'edit', iconName: 'edit', text: 'Edit' },
+  { type: 'icon-button', id: 'search', text: 'Search' },
+  {
+    type: 'menu-dropdown',
+    id: 'misc',
+    text: 'Misc',
+    items: [
+      { id: 'menu-open', iconName: 'file-open', text: 'Open' },
+      { id: 'menu-upload', iconName: 'upload', text: 'Upload' },
+    ],
+  },
+];
+
+const emptyGroup: ButtonGroupProps.ItemOrGroup[] = [
+  {
+    type: 'group',
+    text: 'Feedback',
+    items: [],
+  },
+];
+
+test('renders all items', () => {
+  const wrapper = renderButtonGroup({ items: items1 });
+
+  expect(wrapper.findItems()).toHaveLength(6);
+  expect(wrapper.findButtonById('edit')).not.toBe(null);
+  expect(wrapper.findMenuById('misc')).not.toBe(null);
+});
+
+test('renders stub icon when no icon specified', () => {
+  const wrapper = renderButtonGroup({ items: items1 });
+
+  expect(wrapper.findMenuById('search')?.findAll(`.${buttonStyles.icon}`)).toHaveLength(1);
+});
+
+test('handles menu click event correctly', () => {
+  const onItemClick = jest.fn();
+  const wrapper = renderButtonGroup({ items: items1, onItemClick });
+  const buttonDropdown = wrapper.findMenuById('misc')!;
+  buttonDropdown.openDropdown();
+  buttonDropdown.findItemById('menu-open')!.click();
+
+  expect(onItemClick).toHaveBeenCalledWith(expect.objectContaining({ detail: { id: 'menu-open' } }));
+});
+
+describe('focus', () => {
+  test('focuses the correct item', () => {
+    const ref: { current: ButtonGroupProps.Ref | null } = { current: null };
+    const wrapper = renderButtonGroup({ items: items1 }, ref);
+    ref.current?.focus('copy');
+
+    expect(wrapper.findButtonById('copy')!.getElement()).toHaveFocus();
+  });
+
+  test('focuses on show more button', () => {
+    const ref: { current: ButtonGroupProps.Ref | null } = { current: null };
+    const wrapper = renderButtonGroup({ items: items1 }, ref);
+    ref.current?.focus('misc');
+
+    expect(wrapper.findMenuById('misc')!.getElement().getElementsByTagName('button')[0]).toHaveFocus();
+  });
+
+  test('focuses the correct item with keyboard', async () => {
+    const ref: { current: ButtonGroupProps.Ref | null } = { current: null };
+    const wrapper = renderButtonGroup({ items: items1 }, ref);
+    ref.current?.focus('copy');
+
+    fireEvent.keyDown(wrapper.getElement(), { keyCode: KeyCode.right });
+    await timeout();
```

Comment:
Why do we need the timeouts? I assume it's because of waiting for the react updates. If so, why not using `act` instead?

Reply:
Don't need it anymore. Removed